internal/pkg/cli/task_run.go

```diff
@@ -501,6 +516,30 @@ func (o *runTaskOpts) confirmSecretsAccess() error {
 	return nil
 }
 
+func (o *runTaskOpts) validateEnvCompatibilityForGenerateJobCmd(app, env string) error {
+	envStack, err := o.envCompatibilityChecker(app, env)
+	if err != nil {
+		return err
+	}
+	version, err := envStack.Version()
+	if err != nil {
+		return fmt.Errorf("retrieve version of environment stack %q in application %q: %v", env, app, err)
+	}
+	// The '--generate-cmd' flag was introduced in env v1.4.0. In env v1.8.0,
+	// EnvManagerRole took over, but "states:DescribeStateMachine" permissions
+	// weren't added until 1.12.2. Any version predating the feature's
+	// introduction or between v1.8.0 and 1.12.2 are invalid.
+	if semver.Compare(version, "v1.12.2") < 0 {
+		return &errFeatureIncompatibleWithEnvironment{
+			ws:             o.ws,
```

Comment:
It seems like `ws` here is to list all the local environments and then prompt users to upgrade them. However, `version` here is just one environment. One out-dated environment doesn't necessarily mean they need to upgrade the other environments 🤔 

Sorry this is more like a problem for `errFeatureIncompatibleWithEnvironment`. Since we'll use it I think it's worth bringing it up here.